client/src/pages/private/CohortDetails.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import dayjs from 'dayjs';
+import { Box, Card, Grid, Typography, Link } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+
+import { Page, CheckPermissions } from '../../components/generic';
+import { useToast } from '../../hooks';
+import { fetchCohort, fetchParticipantPostHireStatus } from '../../services';
+import { Routes, ToastStatus, postHireStatuses } from '../../constants';
+
+const useStyles = makeStyles((theme) => ({
+  cardRoot: {
+    minWidth: '1020px',
+  },
+  gridRoot: {
+    padding: theme.spacing(2),
+  },
+}));
+
+export default ({ match }) => {
+  const cohortId = parseInt(match.params.id);
+  const classes = useStyles();
+  const [cohort, setCohort] = useState(null);
+  const [unsuccessfulParticipantsNumber, setUnsuccessfulParticipantsNumber] = useState(-1);
+  const { openToast } = useToast();
+
+  const getTotalParticipantsNumber = () => {
+    return cohort ? cohort.cohort_size : 0;
+  };
+
+  const getAvailableParticipantsNumber = () => {
+    if (!cohort) return 0;
+    const totalNumber = cohort.cohort_size;
+    const assignedParticipantsNumber = cohort.participants?.length;
+    return totalNumber - assignedParticipantsNumber;
+  };
```

Comment:
```suggestion
  const cohortSize = cohort?.cohort_size || 0;
  const assignedParticipants = cohort?.participants.length || 0;
  const availableCohortSeats = cohortSize - assignedParticipants;
```

These functions can be replaced by a few variables, I also adjusted their names to be a bit more reflective of their data

The benefit here is that these variables can be used freely by other parts of the component rather than being nested in a function, as well as just being simpler than creating a function